sky/spot/controller.py

```diff
@@ -145,12 +145,15 @@ def _run(self):
                             'cluster is healthy. Try to recover the job '
                             '(the cluster will not be restarted).')
 
-            resources = list(self._task.resources)[0]
-            if resources.need_cleanup_after_preemption():
-                # Some spot resource (e.g., Spot TPU VM) may need to be
-                # cleaned up after preemption.
-                logger.info('Cleaning up the preempted spot cluster...')
-                self._strategy_executor.terminate_cluster()
+            # When the handle is None, the cluster should be cleaned up already.
```

Comment:
cc @WoosukKwon: this seems like a good example of the Resources refactoring proposal that pushes for clearer distinctions between different uses.